app/backend/prepdocslib/searchmanager.py

```diff
@@ -65,140 +67,190 @@ def __init__(
         self.search_images = search_images
 
     async def create_index(self, vectorizers: Optional[List[VectorSearchVectorizer]] = None):
-        logger.info("Ensuring search index %s exists", self.search_info.index_name)
+        logger.info("Checking whether search index %s exists...", self.search_info.index_name)
 
         async with self.search_info.create_search_index_client() as search_index_client:
-            fields = [
-                (
-                    SimpleField(name="id", type="Edm.String", key=True)
-                    if not self.use_int_vectorization
-                    else SearchField(
-                        name="id",
+
+            if self.search_info.index_name not in [name async for name in search_index_client.list_index_names()]:
```

Comment:
This code change looks larger than it is, because I moved all the index creation logic inside the if statement. Before, we would create the index object even if we didn't use it, which seemed an odd thing to do!